providers-sdk/v1/inventory/asset.go

```diff
@@ -178,3 +178,12 @@ func (a *Asset) AddMondooLabels(root *Asset) {
 		}
 	}
 }
+
+// SetPlatform is an accessor to correctly set the platform of the asset.
+func (a *Asset) SetPlatform(pf *Platform) {
```

Comment:
I think semantically, SetPlatform should override the data. If the caller wants to merge, the caller should call MergePlatform().